web/source/kmwdom.ts

```diff
@@ -720,15 +676,20 @@ class DOMManager {
    * @return      {Array<Element>}        A list of potentially-editable controls.  Further filtering [as with isKMWInput() and
    *                                      isKMWDisabled()] is required.
    */
-  _GetDocumentEditables(Pelem: HTMLElement|Document) {
+  _GetDocumentEditables(Pelem: HTMLElement|Document): (HTMLElement|Document)[] {
     var util = this.keyman.util;
 
-    var possibleInputs = [];
+    var possibleInputs: (HTMLElement|Document)[] = [];
 
-    if(Util.instanceof(Pelem, "HTMLInputElement") || Util.instanceof(Pelem, "HTMLTextAreaElement")) {
-      possibleInputs.push(Pelem);
-    } else if(Util.instanceof(Pelem, "HTMLIFrameElement")) {
-      possibleInputs.push(Pelem);
+    // Document.ownerDocument === null, so we better check that it's not null before proceeding.
+    if(Pelem.ownerDocument && Pelem instanceof Pelem.ownerDocument.defaultView.HTMLElement) {
+      var dv = Pelem.ownerDocument.defaultView;
+
+      if(Pelem instanceof dv.HTMLInputElement || Pelem instanceof dv.HTMLTextAreaElement) {
+        possibleInputs.push(Pelem);
+      } else if(Pelem instanceof dv.HTMLIFrameElement) {
+        possibleInputs.push(Pelem);
+      }
     } else if(Pelem.nodeName == "#text") {
```

Comment:
Is this even possible if `Pelem` is a `HTMLElement` and not a `Node`?

Reply:
Probably not; it's inherited there from legacy stuffs, and I was hesitant to cut it b/c of the current HTMLElement/Document ambiguities to be resolved in the future.  It's probably safe to kill, so sure, I'll nix it.